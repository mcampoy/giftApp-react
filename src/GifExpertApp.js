import React, {useState} from 'react';
import AddCategory  from './components/AddCategory';
import Footer from './components/Footer';
import GifGrid from './components/GifGrid';


const GifExpertApp = ({ defaultCategories = [] }) => {

    // const [categories, setCategories] = useState([ 'cats' ]);
    const [categories, setCategories] = useState( defaultCategories );

    return (
        <>
            <h2>Busc√° tus GIFs favoritos</h2>
            <AddCategory setCategories={setCategories}/>
            <hr />
            <div className="wrapper">
                {
                    categories.map(category => (
                        <GifGrid 
                        key={category}
                        category={ category } /> 
                    ))
                }
            </div>
            <Footer />
        </>
    )

}

export default GifExpertApp;