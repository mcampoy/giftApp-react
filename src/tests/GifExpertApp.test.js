import React  from 'react';
import GifExpertApp from '../GifExpertApp';
const { shallow } = require("enzyme");


describe('Se realizan pruebas en <GifExpertApp />', () => {

    test('debe mostrar correctamente el comoponente', () => {

        const wrapper = shallow(<GifExpertApp/>);

        expect( wrapper ).toMatchSnapshot();

    })

    test('debe mostrar una lista de categorÃ­as', () => {
        
        const categories = [ 'cats', 'dogs' ];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length )


    })



})