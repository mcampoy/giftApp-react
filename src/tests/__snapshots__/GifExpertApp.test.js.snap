// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Se realizan pruebas en <GifExpertApp /> debe mostrar correctamente el comoponente 1`] = `
<Fragment>
  <h2>
    Buscá tus GIFs favoritos
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="wrapper"
  />
  <Footer />
</Fragment>
`;

exports[`Se realizan pruebas en <GifExpertApp /> debe mostrar una lista de categorías 1`] = `
<Fragment>
  <h2>
    Buscá tus GIFs favoritos
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div
    className="wrapper"
  >
    <GifGrid
      category="cats"
      key="cats"
    />
    <GifGrid
      category="dogs"
      key="dogs"
    />
  </div>
  <Footer />
</Fragment>
`;
